webpack.config.js

```diff
@@ -1,86 +1,88 @@
-        test: /\.js?$/,
-        include: path.resolve(__dirname, './src/charts'),
-        exclude: /(node_modules)/,
-        enforce: 'pre',
-        loader: 'eslint-loader',
+const webpack = require('webpack');
+const path = require('path');
+const LiveReloadPlugin = require('webpack-livereload-plugin');
+const UglifyJsPlugin = webpack.optimize.UglifyJsPlugin;
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+const merge = require('webpack-merge');
+
+const env = require('yargs').argv.mode;
+const isProduction = env === 'prod' || env === 'prodUMD';
+
+const projectName = 'britecharts';
+
+const CHARTS = {
+    'bar': './src/charts/bar.js',
+    'donut': './src/charts/donut.js',
+    'legend': './src/charts/legend.js',
+    'line': './src/charts/line.js',
+    'tooltip': './src/charts/tooltip.js',
+    'miniTooltip': './src/charts/mini-tooltip.js',
+    'sparkline': './src/charts/sparkline.js',
+    'stackedArea': './src/charts/stacked-area.js',
+    'stackedBar': './src/charts/stacked-bar.js',
+    'groupedBar': './src/charts/grouped-bar.js',
+    'step': './src/charts/step.js',
+    'brush': './src/charts/brush.js',
+    'loading': ['./src/charts/helpers/load.js'],
+    // hack to make webpack use colors as an entry point while its also a dependency of the charts above
+    'colors': ['./src/charts/helpers/color.js']
+};
+const DEMOS = {
+    'demo-line': './demos/demo-line.js',
+    'demo-stacked-area': './demos/demo-stacked-area.js',
+    'demo-bar': './demos/demo-bar.js',
+    'demo-grouped-bar': './demos/demo-grouped-bar.js',
+    'demo-stacked-bar': './demos/demo-stacked-bar.js',
+    'demo-donut': './demos/demo-donut.js',
+    'demo-sparkline': './demos/demo-sparkline.js',
+    'demo-step': './demos/demo-step.js',
+    'demo-brush': './demos/demo-brush.js',
+    'demo-kitchen-sink': './demos/demo-kitchen-sink.js'
+};
+const PATHS = {
+    vendor: path.resolve('./node_modules'),
+    bundleIndex: path.resolve('./src/bundle.js'),
+    charts: path.resolve('./src/charts'),
+};
+
+// Move to Parts
+const babelLoader = {
+    test: /\.js$/,
+    exclude: /(node_modules)/,
+    use: [{
+        loader: 'babel-loader',
         options: {
-            emitWarning: true,
-            failOnError: false,
+            presets: ['es2015', 'stage-0'],
+            cacheDirectory: true,
         }
-    },
+    }],
+
+};
+const babelIstambulLoader = {
+    test: /\.js?$/,
+    include: /src/,
+    exclude: /(node_modules|__tests__|tests_index.js)/,
+    use: [{
+        loader: 'istanbul-instrumenter-loader',
+        options:  {
+            presets: ['es2015', 'stage-0'],
+            cacheDirectory: true,
+        },
+    }],
+};
+const lintJSLoader = {
+    test: /\.js?$/,
+    include: PATHS.charts,
+    exclude: /(node_modules)/,
+    enforce: 'pre',
+    loader: 'eslint-loader',
+    options: {
+        emitWarning: true,
+        failOnError: false,
+    }
+};
 
-    plugins = [
+let plugins = [
```

Comment:
Why `let` for this one?

Reply:
In line 91, if the environment is production, we push something in.

That said, a const won't break if we do that, but it feels weird.

Reply:
I see, so this conveys intent to change the values inside of the array.

Reply:
I think it is a good convention to declare variables with `let` if they are being mutated in the future (unless they are 3rd party modules or a custom module of some sort.)